src/nhs_grid.jl

```diff
@@ -86,37 +87,28 @@ struct GridNeighborhoodSearch{NDIMS, ELTYPE, CL, PB} <: AbstractNeighborhoodSear
             end
         end
 
-        new{NDIMS, ELTYPE, typeof(cell_list),
-            typeof(periodic_box)}(cell_list, search_radius, periodic_box, n_cells,
-                                  cell_size, cell_buffer, cell_buffer_indices,
-                                  threaded_update)
+        new{NDIMS, ELTYPE, typeof(cell_list), typeof(periodic_box),
+            typeof(update_buffer)}(cell_list, search_radius, periodic_box, n_cells,
+                                   cell_size, update_buffer, threaded_update)
     end
 end
 
 @inline Base.ndims(::GridNeighborhoodSearch{NDIMS}) where {NDIMS} = NDIMS
 
-@inline function npoints(neighborhood_search::GridNeighborhoodSearch)
-    return size(neighborhood_search.cell_buffer, 1)
-end
-
 function initialize!(neighborhood_search::GridNeighborhoodSearch,
                      x::AbstractMatrix, y::AbstractMatrix)
     initialize_grid!(neighborhood_search, y)
 end
 
-function initialize_grid!(neighborhood_search::GridNeighborhoodSearch{NDIMS},
-                          y::AbstractMatrix) where {NDIMS}
-    initialize_grid!(neighborhood_search, i -> extract_svector(y, Val(NDIMS), i))
-end
-
-function initialize_grid!(neighborhood_search::GridNeighborhoodSearch, coords_fun)
```

Comment:
Initialization via a coordinate function was a feature that I kept because "why not?", but now it's in the way, and it's not used anyway, so I removed it.